### PR TITLE
Move IPeaksWorkspace properties -> PeaksWorkspace properties

### PR DESCRIPTION
**Pull request by samueljackson92**
**Description of work.**

TL;DR: This makes all algorithms that take peaks workspaces use `PeaksWorkspace` and not `IPeaksWorkspace` and exports a new workspace property (`PeaksWorkspaceProperty`) to python.

---

There's an inconsistency with the code base where in some places we use workspace properties that use `IPeaksWorkspace` and some that use `PeaksWorkspace`. This normally doesn't matter too much until you try and chain algorithms together in python. 

Before this PR we didn't have a `PeaksWorkspaceProperty` so algorithms that use `PeaksWorkspace` as output properties could be marked as `StoreInADS=False` because there was no way to get the value from the property.

Adding this causes a bigger problem because now we have a mix of algorithms that have `IPeaksWorkspace` properties and ones which have `PeaksWorkspace` properties. Grepping the code base suggests that there are many more algorithms that require `PeaksWorkspace` compare to `IPeaksWorkspace` and some of the algorithms which do use `IPeaksWorkspace` actually downcast to a `PeaksWorkspace` anyway.

**To test:**
 - Check all unit & system tests are passing
 - Code review

Does this update require release notes?
- [ ] Yes
- [x] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by soininen — CHANGES_REQUESTED** [review]
The changes look sensible and tests pass. Just a tiny formatting comment below.

**Review by soininen — APPROVED** [review]
